src/google-classroom-extractor/google_classroom_extractor/api/courses.py

```diff
@@ -12,19 +12,65 @@
 
 
 def request_courses(resource: Optional[Resource]) -> List[Dict[str, str]]:
+    """
+    Fetch Course API data for all courses and return a list of course data
+
+    Parameters
+    ----------
+    resource: Optional[Resource]
+        a Google Classroom SDK Resource
+
+    Returns
+    -------
+    List[Dict[str, str]]
+        a list of Google Classroom Course resources,
+            see https://developers.google.com/classroom/reference/rest/v1/courses
+    """
+
     assert isinstance(resource, Resource) or resource is None
 
     if resource is None:
         return []
 
     return call_api(
         resource.courses().list,  # type: ignore - courses() is dynamic
```

Comment:
mypy complains about this being an invalid ignore. When I remove it, Pylance does _not_ complain. Same on the next line.

Reply:
fixed for mypy.  pylance does complain, looking into.